### DISCUSSION TITLE
generateAuthLink throwing 404 error

### DISCUSSION THREAD

**Discussion by zeentraland** [plhery/node-twitter-api-v2#280] [General]
Trying to generate an auth link for 3 legged OAuth1 login, but getting 404 error everytime

`
ApiResponseError: Request failed with code 404
    at RequestHandlerHelper.createResponseError (E:\twitter-auth-app\twitter-api-v2-user-oauth-flow-example\node_modules\twitter-api-v2\dist\client-mixins\request-handler.helper.js:99:16)
    at RequestHandlerHelper.onResponseEndHandler (E:\twitter-auth-app\twitter-api-v2-user-oauth-flow-example\node_modules\twitter-api-v2\dist\client-mixins\request-handler.helper.js:248:25)
    at Gunzip.emit (events.js:315:20)
    at endReadableNT (internal/streams/readable.js:1327:12)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  error: true,
  type: 'response',
  code: 404,
  `

Simply cloned the https://github.com/alkihis/twitter-api-v2-user-oauth-flow-example repo, and set the consumer keys and getting this error. 

Also tried the generateAuthLink method in isolation outside the context of the above example here as shown in the attached image, but still getting the same error. 
![image](https://user-images.githubusercontent.com/95356780/167307945-e71b5bb2-56dd-4ca9-9e06-2d5017576a71.png)

Any likely reason for this error? @alkihis

**Comment by anuragk15** [comment]
Facing the same issue. It was working fine a month ago, looks like twitter has changed its URL endpoint.

**Comment by alkihis** [comment]
Hi,
It works for me for the exact same URL.

```ts
function getRequestClient() {
  return new TwitterApi({
    appKey: process.env.CONSUMER_TOKEN!,
    appSecret: process.env.CONSUMER_SECRET!,
  });
}

const options = await getRequestClient().generateAuthLink('http://127.0.0.1:5000/callback') 
console.log(options)
```

shows:
```ts
{
  url: 'https://api.twitter.com/oauth/authenticate?oauth_token=xxx',
  oauth_token: 'xxx',
  oauth_token_secret: 'xxx',
  oauth_callback_confirmed: 'true'
}
```

Try to print the full error to know whats happening.
```ts
} catch (err) {
  console.log(err, err.stack, err.toJSON?.())
}
```

**Comment by zeentraland** [comment]
Seems to be related to this issue at the API end
https://twittercommunity.com/t/getting-404-for-https-api-twitter-com-oauth-request-token/170925

**Reply by zeentraland** [reply]
https://twittercommunity.com/t/error-when-authorize-twitter-nothing-to-see-here/170967/3

**Reply by alkihis** [reply]
It is only happening on Windows OSes right?

**Reply by anuragk15** [reply]
> It is only happening on Windows OSes right?
Not really. 
I tried on Linux and Mac and had the same issue.
Happening across India and Japan.

**Comment by anuragk15** [comment]
The solution so far that worked for me was using a VPN. The server was based in Germany where the error hasn't occurred(yet).

**Reply by alkihis** [reply]
Thank you for the details.
Using a VPN should be a good solution for now.
With a quick test on my side now, it's working on my servers (Germany/UK) and on my own connection (France).
